pkg/dumper/dump.go

```diff
@@ -19,7 +25,65 @@ type Uploader interface {
 	Upload(input *s3manager.UploadInput, options ...func(*s3manager.Uploader)) (*s3manager.UploadOutput, error)
 }
 
-func New(uploader Uploader, bucket string, logger *zap.Logger) S3DumpClient {
+type RoundRobinDumpClient struct {
+	logger  *zap.Logger
+	clients []Dumper
+}
+
+func NewRoundRobinDumpClient(logger *zap.Logger, clients ...Dumper) RoundRobinDumpClient {
+	return RoundRobinDumpClient{
+		logger,
+		clients,
+	}
+}
+
+func (c RoundRobinDumpClient) Dump(ctx context.Context, r io.Reader, path string) error {
+	b, err := ioutil.ReadAll(r)
```

Comment:
This could be dangerous with a large input. It's probably fine for now, but it's something to be mindful of in the future.